### ISSUE TITLE
VoiceOver doesn't announce tooltip inside dialog

### ISSUE THREAD

**Issue by shamelio** [floating-ui/floating-ui#3089]
**Describe the bug** A clear and concise description of what the bug is.

**To Reproduce**

1. Go to https://codesandbox.io/p/sandbox/hardcore-glitter-dqkvc7
2. Open VoiceOver
3. Click "My Trigger" button
4. Tooltip is opened but VoiceOver doesn't announce its content

**Expected behavior** 
VoiceOver / SR should announce the tooltip content

**Screenshots** 
![image](https://github.com/user-attachments/assets/904d20da-d546-4e15-882e-ad897e9e249c)

**Context:**

- OS: MacOs / iOs
- Browser: chrome
- Version: 0.26.24

**Additional context** 
I've copied the dialog and tooltip react examples and only added `aria-modal` to the dialog.
I've tried to conditionally set the tooltip trigger `aria-describedby` when `open` is true - didn't work.
When tooltip `open` is always true - it works.

**Comment by atomiks** [comment]
Secondary roles like `tooltip` aren't guaranteed to be announced; a `label` role should instead be used.

Also, this isn't what the example shows, but I now recommend making all tooltips visual-only and adding `aria-label` on the button instead if it's an icon button. Lastly, VoiceOver is not consistent or reliable with how it operates and is pretty buggy in general.
